In [1]:
%load_ext tikzmagic
%matplotlib inline

# 탐색적 데이터 분석

EDA(Exploratory Data Analysis)라고도 부르는 탐색적 데이터 분석은 데이터 각각에 대한 특징을 찾아내는 작업이다. 탐색적 데이터 분석에서 찾아내야 할 데이터의 특징은 여러가지가 있지만 가장 중요한 것은 해당 데이터가 어떤 확률분포를 따르며 확률분포함수의 모수가 어떤 값을 가지는 지를 추정하는 것이다.

탐색적 데이터 분석에서 수행해야 할 일은 다음과 같다.

* 데이터의 개수
* 데이터 중 누락 혹은 잘못된 데이터의 개수
* 데이터가 수치형 데이터인지 범주형 데이터인지 유형 판단
* 데이터의 대표값 (수치형 데이터의 경우에는 평균이나 중간값, 범주형 데이터의 경우에는 최빈값)
* 데이터의 최대/최소값 혹은 분산
* 데이터의 제한조건 (수치형 데이터의 경우에는 제한조건, 범주형 데이터의 경우에는 범주값의 종류)
* 데이터의 확률분포 및 모수


## 탐색적 데이터 분석의 예 : 팁 데이터

지금까지 많이 살펴보았던 팁 데이터에 대해 탐색적 데이터 분석을 하면 다음과 같은 결과를 얻을 수 있다.

In [2]:
import seaborn as sns

tips = sns.load_dataset("tips")
tips

total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
1         10.34  1.66    Male     No   Sun  Dinner     3
2         21.01  3.50    Male     No   Sun  Dinner     3
3         23.68  3.31    Male     No   Sun  Dinner     2
4         24.59  3.61  Female     No   Sun  Dinner     4
..          ...   ...     ...    ...   ...     ...   ...
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 7 columns]

데이터의 개수나 평균, 최대/최소값 등은 pandas 데이터프레임의 `describe` 메서드로 얻을 수 있다.

In [3]:
tips.describe()

total_bill         tip        size
count  244.000000  244.000000  244.000000
mean    19.785943    2.998279    2.569672
std      8.902412    1.383638    0.951100
min      3.070000    1.000000    1.000000
25%     13.347500    2.000000    2.000000
50%     17.795000    2.900000    2.000000
75%     24.127500    3.562500    3.000000
max     50.810000   10.000000    6.000000

탐색적 데이터 분석의 결과를 정리하면 다음과 같다.


| 이름 | 개수 | 누락/오류 데이터 개수 | 데이터 유형 | 데이터 대표값 | 최대/최소값 | 데이터 제한조건 | 데이터 확률분포 |
|-|-|-|-|-|-|-|-|
| total_bill | 243 | 0 | 수치형 | 19.7859 | 3.07 / 50.81 | 양수 | 로그정규분포 |
| tip | 243 | 0 | 수치형 | 2.9983 | 1 / 10 | 양수 | 로그정규분포 | 
| size | 243 | 0 | 수치형 | 2.5697 | 1 / 6 | 양수, 정수 | 정규분포 | 